### PR TITLE
Editing status flow

### PR DESCRIPTION
**Pull request by sallamTanna**
We've edit some logic in status process.
- The Pending orders have no due date.
- The Approved orders have 'Approved date'.
- The Delivered orders have 'Delivered date', this date appears only when the user has entered the tracking number.

### CODE REVIEW

File: src/server/model/queries/select.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const connect = require('../db_connect.js');
 
 exports.selecAllOrders = (cb) => {
-  const sql = 'select i.title, o.id, o.quantity, o.seen ,o.status, o.delivery_time, i.image from orders o inner join items i on o.item_id=i.id;';
+  const sql = 'select i.title, o.id, o.quantity, o.traking_number, o.approved_date, o.seen ,o.status, o.delivery_date, i.image from orders o inner join items i on o.item_id=i.id;';
 
   connect.query(sql, (err, result) => {
     if (err) return cb(new Error(err));
```
Comment (Eslam3bed) [inline src/server/model/queries/select.js]: no need to create new Error instance just `cb(err)` will do the job

File: src/server/controllers/index.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ router.post('/signup', signUp);
 router.get('/getAllOrders', getAllOrders.getAllOrders);
 router.get('/getAllItems', getAllItems.getAllItems);
 router.post('/updateSeenValue', updateSeenValue.updateSeenValue);
-router.post('/changeOrderStatus', changeOrderStatus.changeOrderStatus);
+router.post('/changeOrderStatusAndApprovedDate', changeOrderStatus.changeOrderStatus);
```
Comment (Eslam3bed) [inline src/server/controllers/index.js]: routes names should be very abstract
avoid longe names
you could do somthing like : 
```js
router.post('/order/:id/state')
router.update('/order/:id') 
// or
router.post('/order/state') // if you recognize the order id from the body
```
 eventually it's your call to decide which paradigm you take

File: src/server/model/queries/select.js (hunk 1 above)
Comment (sallamTanna) [reply src/server/model/queries/select.js]: Edited.

File: src/server/controllers/index.js (hunk 2 above)
Comment (sallamTanna) [reply src/server/controllers/index.js]: Edited.

Comment (sallamTanna) [reply src/server/controllers/index.js]: Edited.

**Pull request merged**